apis/python/src/tiledb/vector_search/flat_index.py

```diff
@@ -80,10 +86,96 @@ def query_internal(
         # TODO:
         # - typecheck queries
         # - add all the options and query strategies
+        if self.size == 0:
+            return np.full((queries.shape[0], k), index.MAX_FLOAT_32), np.full((queries.shape[0], k), index.MAX_UINT64)
 
         assert queries.dtype == np.float32
 
         queries_m = array_to_matrix(np.transpose(queries))
         d, i = query_vq_heap(self._db, queries_m, self._ids, k, nthreads)
 
         return np.transpose(np.array(d)), np.transpose(np.array(i))
+
+
+def create(
+    uri: str,
+    dimensions: int,
+    vector_type: np.dtype,
+    config: Optional[Mapping[str, Any]] = None,
+    **kwargs
+) -> FlatIndex:
+    with tiledb.scope_ctx(ctx_or_config=config):
+        try:
+            tiledb.group_create(uri)
+        except tiledb.TileDBError as err:
+            message = str(err)
+            if "already exists" not in message:
+                raise err
+        group = tiledb.Group(uri, "w")
+        group.meta["dataset_type"] = index.DATASET_TYPE
+        group.meta["dtype"] = np.dtype(vector_type).name
+        group.meta["storage_version"] = STORAGE_VERSION
+        group.meta["index_type"] = INDEX_TYPE
+        group.meta["base_sizes"] = json.dumps([0])
+        group.meta["ingestion_timestamps"] = json.dumps([0])
+        tile_size = TILE_SIZE_BYTES / np.dtype(vector_type).itemsize / dimensions
+
+        ids_array_name = storage_formats[STORAGE_VERSION]["IDS_ARRAY_NAME"]
+        parts_array_name = storage_formats[STORAGE_VERSION]["PARTS_ARRAY_NAME"]
+        ids_uri = f"{uri}/{ids_array_name}"
+        parts_uri = f"{uri}/{parts_array_name}"
+
+        if not tiledb.array_exists(ids_uri):
```

Comment:
Should we error out if the array already exists, since this is a `create` function?